tests/api2/pool_dataset_encryption.py

```diff
@@ -0,0 +1,870 @@
+    assert results.status_code == 200, results.text
+
+
+def test_06_create_not_encrypted_dataset_on_a_normal_pool():
+    payload = {
+        'name': dataset,
+        'encryption': False,
+    }
+    results = POST('/pool/dataset/', payload)
+    assert results.status_code == 200, results.text
+
+
+def test_07_delete_not_encrypted_dataset():
+    results = DELETE(f'/pool/dataset/id/{dataset_url}/')
+    assert results.status_code == 200, results.text
+
+
+def test_08_create_a_dataset_with_inherit_encryption_true_on_a_normal_pool():
+    payload = {
+        'name': dataset,
+        'inherit_encryption': True
+    }
+    results = POST('/pool/dataset/', payload)
+    assert results.status_code == 200, results.text
+
+
+def test_09_delete_dataset():
+    results = DELETE(f'/pool/dataset/id/{dataset_url}/')
+    assert results.status_code == 200, results.text
+
+
+def test_10_try_to_create_an_encrypted_dataset_with_pbkdf2itersl_zero():
+    payload = {
+        'name': dataset,
+        'encryption_options': {
+            'pbkdf2iters': 0,
+        },
+        'encryption': True,
+        'inherit_encryption': False
+    }
+    results = POST('/pool/dataset/', payload)
+    assert results.status_code == 422, results.text
+    assert 'Should be greater or equal than 100000' in results.text, results.text
+
+
+def test_11_try_to_create_an_encrypted_dataset_with_inherit_encryption_true():
+    payload = {
+        'name': dataset,
+        'encryption_options': {
+            'passphrase': 'my_passphrase',
+        },
+        'encryption': True,
+        'inherit_encryption': True
+    }
+    results = POST('/pool/dataset/', payload)
+    assert results.status_code == 422, results.text
+    assert 'Must be disabled when encryption is enabled' in results.text, results.text
+
+
+def test_12_try_to_create_an_encrypted_dataset_with_passphrase_and_generate_key():
+    payload = {
+        'name': dataset,
+        'encryption_options': {
+            'generate_key': True,
+            'passphrase': 'my_passphrase',
+        },
+        'encryption': True,
+        'inherit_encryption': False
+    }
+    results = POST('/pool/dataset/', payload)
+    assert results.status_code == 422, results.text
+    assert 'Must be disabled when dataset is to be encrypted with passphrase' in results.text, results.text
+
+
+def test_13_create_an_encrypted_root_with_generate_key():
+    payload = {
+        'name': dataset,
+        'encryption_options': {
+            'generate_key': True,
+        },
+        'encryption': True,
+        'inherit_encryption': False
+    }
+    results = POST('/pool/dataset/', payload)
+    assert results.status_code == 200, results.text
+
+
+@skip_for_ha
+def test_14_delete_generate_key_encrypted_root():
+    results = DELETE(f'/pool/dataset/id/{dataset_url}/')
+    assert results.status_code == 200, results.text
+
+
+@skip_for_ha
+def test_15_create_an_encrypted_root_with_generate_key():
```

Comment:
Please correct the function name to reflect the reality of the operation being performed